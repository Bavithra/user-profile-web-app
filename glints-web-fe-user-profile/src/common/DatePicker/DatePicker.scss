.custom-style {
  .react-datepicker-popper[data-placement^="top"] .react-datepicker__triangle,
  .react-datepicker-popper[data-placement^="bottom"]
    .react-datepicker__triangle {
  }
}
