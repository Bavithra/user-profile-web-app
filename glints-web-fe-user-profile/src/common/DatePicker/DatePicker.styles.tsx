import styled from 'styled-components';

import { Colors } from '../../styles/Colors';
import { buttonStyle } from '../../styles/Common.styles';

export const HeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const ToggleContainer = styled.div`
  display: flex;
  margin: 0 10px;
`;

export const ArrowButton = styled.div`
  background: ${Colors.purewhite};
`;

export const ChildrenContainer = styled.div`
  display: flex;
  justify-content: center;
`;

export const Container = styled.div`
  margin: 4px 0;
  padding: 0 12px;
  font-size: 14px;
  color: ${Colors.grey2};
`;

export const InputContainer = styled.div`
  width: 100%;
`;

export const IconBackground = styled.div`
  position: absolute;
  height: 40px;
  width: 48px;
  background: ${Colors.primary1};
  box-shadow: 0px 3px 6px #00000029;
  right: 0;
  top: 0;

  ${buttonStyle}
`;
