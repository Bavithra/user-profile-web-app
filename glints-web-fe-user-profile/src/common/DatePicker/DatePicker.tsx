import React, { useState } from "react";
import ReactDatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import DateTimeUtil from "../../utils/DateTimeUtil";

import TextInput from "../TextInput";
import { DATE_FORMAT } from "../../Constants";
import { Container } from "./DatePicker.styles";

type Props = {
  selectedDate: Date | undefined;
  placeholder: string;
  onChange: (date: Date) => void;
  isMonthPicker?: boolean;
  dateFormat?: string;
};

export default function DatePicker(props: Props) {
  const {
    selectedDate,
    placeholder,
    isMonthPicker = false,
    dateFormat = DATE_FORMAT,
    onChange,
  } = props;
  const [isOpen, setIsOpen] = useState<boolean>(false);

  function handleClick() {
    setIsOpen(!isOpen);
  }

  function handleSelectedDateChange(value: string) {
    onChange(DateTimeUtil.toDate(value)!);
  }

  return (
    <ReactDatePicker
      open={isOpen}
      preventOpenOnFocus={true}
      onChange={onChange}
      onClickOutside={handleClick}
      selected={selectedDate}
      onSelect={handleClick}
      dateFormat={dateFormat}
      showMonthYearPicker={isMonthPicker}
      showYearDropdown
      maxDate={new Date()}
      customInput={
        <Container>
          <div onClick={handleClick}>
            {selectedDate ? (
              <TextInput
                readOnly={isMonthPicker}
                label={placeholder}
                value={DateTimeUtil.getDisplayDateWithDay(
                  selectedDate,
                  dateFormat
                )}
                onChange={handleSelectedDateChange}
              />
            ) : (
              <TextInput
                readOnly={isMonthPicker}
                label={placeholder}
                onChange={handleSelectedDateChange}
              />
            )}
          </div>
        </Container>
      }
    />
  );
}
