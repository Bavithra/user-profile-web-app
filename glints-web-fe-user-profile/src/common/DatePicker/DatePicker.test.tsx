import React from "react";
import DatePicker from "./DatePicker";
import renderer from "react-test-renderer";

describe("DatePicker Block", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <DatePicker
          selectedDate={new Date()}
          placeholder={'Select Date'}
          onChange={function (date: Date | null): void {
            throw new Error("Function not implemented.");
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
