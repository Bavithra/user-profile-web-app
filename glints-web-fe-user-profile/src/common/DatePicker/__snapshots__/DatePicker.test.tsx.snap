// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Block renders correctly 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <div
      className="sc-ftvSup laBHLg"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      readOnly={false}
      value=""
    >
      <div
        onClick={[Function]}
      >
        <div
          className="sc-dkzDqf fSUJtn"
        >
          <input
            className="sc-kDDrLX emimmX"
            onChange={[Function]}
            type="type"
          />
          <label
            className="sc-jqUVSM dsSNWu"
          >
            Select Date
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
